SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -157,6 +143,36 @@ def update(session, args):
             raise error
         return json.dumps(res)
 
+@error_wrapped
+@operationlock(timeout=10)
+def install(session, args):
+    if not args.get('packages'):
+        raise Exception('`packages` is empty')
+    return install_helper(session, args, 'install')
+
+@error_wrapped
+@operationlock()
+def check_update(session, args):
+    yum_instance = yum.YumBase()
+    yum_instance.preconf.debuglevel = 0
+    yum_instance.preconf.plugins = False
+    repos = build_repo_list(yum_instance.repos.listEnabled(), args.get('repos'))
+    yum_instance.preconf.plugins = True
```

Comment:
I thought you told me that `preconf` can't be used after calling `listEnabled()`?

Reply:
Yeah I have this change locally. You have a good memory.